data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/reconciler/impl/ResourcesReconcilerImpl.java

```diff
@@ -75,67 +75,11 @@ private ResourcesReconcilerImpl(
         // Update to new resource
         resourceEntry.setValue(newResource);
 
-        boolean resourceAreEquals = resourceEquals(oldResource, newResource);
+        final boolean resourceAreEquals = resourceEquals(oldResource, newResource);
 
-        // Reconcile ingress
-        if (isReconcilingIngress() && !resourceAreEquals) {
-          if (!oldResource.hasIngress() && newResource.hasIngress()) {
-            futures.add(
-              this.ingressReconcilerListener.onNewIngress(newResource, newResource.getIngress())
-            );
-          } else if (oldResource.hasIngress() && !newResource.hasIngress()) {
-            futures.add(
-              this.ingressReconcilerListener.onDeleteIngress(oldResource, oldResource.getIngress())
-            );
-          } else if (newResource.hasIngress()) {
-            futures.add(
-              this.ingressReconcilerListener.onUpdateIngress(newResource, newResource.getIngress())
-            );
-          }
-        }
+        reconcileIngress(futures, oldResource, newResource, resourceAreEquals);
```

Comment:
Why that change? The linter doesn't like big functions? I don't like that, big functions might have a performance reasoning behind it, i would prefer to remove that lint

Reply:
Cyclomatic complexity was really high, should I remove it? It seems much more readable now.

Reply:
TBH I personally don't like cyclomatic complexity as a measurement of good code. In this case, I rather prefer to have the full algorithm in a single place, more than cutted in several pieces. This way I personally think it increases the brain overload to grasp the full algorithm, because I must go through all the different indirections. Deciding whether we should create a function or not is a human job, not a tooling job.

Also, as a side note, from the JIT point of view, this code might be optimized in a different way than the original one.